modelkit/core/profilers/simple.py

```diff
@@ -0,0 +1,99 @@
+import time
+from collections import defaultdict
+from typing import Dict, Union, Set
+import numpy as np
+import pandas as pd
+
+from modelkit.core.profilers.base import BaseProfiler
+
+
+class SimpleProfiler(BaseProfiler):
```

Comment:
An idea for future PR, we may propose an simple implementation using https://pypi.org/project/prometheus-client/